stan/math/prim/prob/wiener5_lpdf.hpp

```diff
@@ -0,0 +1,637 @@
+#ifndef STAN_MATH_PRIM_PROB_WIENER5_LPDF_HPP
+#define STAN_MATH_PRIM_PROB_WIENER5_LPDF_HPP
+
+#include <stan/math/prim/fun.hpp>
+
+namespace stan {
+namespace math {
+namespace internal {
+
+// calculate density in log
+inline double dwiener5(const double& y, const double& a, const double& vn,
+                       const double& wn, const double& sv, const double& err) {
+  double kll, kss, ans, v, w;
+  w = 1.0 - wn;
+  v = -vn;
+  double y_asq = y / square(a);
+  ans = 0.0;
+
+  // calculate the number of terms needed for short t
+  double lg1;
+  if (sv != 0) {
+    double sv_sqr = square(sv);
+    double one_plus_svsqr_y = 1 + sv_sqr * y;
+    lg1 = (sv_sqr * square(a * w) - 2 * a * v * w - square(v) * y) / 2.0
+              / one_plus_svsqr_y
+          - 2 * log(a) - 0.5 * log(one_plus_svsqr_y);
+  } else {
+    lg1 = (-2 * a * v * w - square(v) * y) / 2.0 - 2 * log(a);
+  }
+  double es = (err - lg1);
+  double K1 = (sqrt(2.0 * y_asq) + w) / 2.0;
+  double u_eps = fmin(-1.0, LOG_TWO + LOG_PI + 2.0 * log(y_asq) + 2.0 * (es));
+  double arg = -y_asq * (u_eps - sqrt(-2.0 * u_eps - 2.0));
```

Comment:
@Franzi2114 would you mind quickly clarifying why the construction of these terms is slightly different here compared to the derivative functions? For example, these quantities are computed in `dtdwiener5` as:
```cpp
  double K1 = (sqrt(3.0 * y_asq) + w) / 2.0;
  double u_eps = fmin(
      -1.0, (log(8.0 / 27.0) + LOG_PI + 4.0 * log(y_asq) + 2.0 * es) / 3.0);
  double arg = -3.0 * y_asq * (u_eps - sqrt(-2.0 * u_eps - 2.0));
```